src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs

```diff
@@ -244,11 +252,11 @@ private void OnWrite()
             {
                 _compressionChecked = true;
 
-                InitializeCompressionHeaders();
+                var compressionProvider = InitializeCompressionHeaders();
 
-                if (_compressionProvider != null)
```

Comment:
Does this leave a field that is never initialized now?

Reply:
This wasn't where it was initialized, only consumed. It's initialized over here:
https://github.com/dotnet/aspnetcore/blob/59baa0d6f93022e87cc82faaf0433fe3513a38a9/src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs#L269

The _compressionProvider field exists because there are two code paths that might cause it to be created, OnWrite and DisableBuffering, and we don't want to create it twice.

Reply:
But now the field is never assigned in OnWrite, so it will be created twice?

Reply:
Ah, I see - it `ResolveCompressionProvider` is called from `InitialializeCompressionHeader`, so it does end up being the same instance.

Reply:
It's assignment hasn't changed, that happens in ResolveCompressionProvider via:
- OnWrite()
  - InitializeCompressionHeaders();
    - ResolveCompressionProvider();

The alternate code path is:
- DisableBuffering()
  - ResolveCompressionProvider();